doc/chplvis/chplvis.rst

```diff
@@ -0,0 +1,315 @@
+  If two locales do not communicate, no line is drawn between them.
+  If communication is only one way, the communication color for
+  *no communication* is gray.
+
+- The *Data menu* controls what data is used for display colors.  This
+  initial data is number of tasks for locales and number of
+  communications calls for the communication links.  For locales, one
+  can select number of tasks, CPU time at the locale, clock time or
+  concurrency at the locale.  Clock time is normally very close to
+  equal across all locales.  For the communication links, one can
+  select number of communications or size of data sent.</li>
+
+Display Interaction
+-------------------
+
+Clicking on elements of the display will bring up more information.
+Clicking on a locale will open a new window for that locale showing
+information for that locale.  For example, in example 1, clicking on
+locale 0 when the locale data is 'number of tasks', 'CPU time' or 'clock
+time' will produce a window that looks like:
+
+.. image:: E1-L0.png
+
+(Note: There is overhead generated in tasks, CPU time and clock
+time for the Visual Debug function calls.  *chplvis* removes
+from the task count the overhead tasks but it can not remove the
+CPU and clock time overhead.)
+
+When the locale data selected is 'concurrency', clicking on a locale
+will bring up a window that shows task on the locale.  This display
+shows the order the tasks are executed and the color of each task
+shows the clock time for that task.   For example, locale 0 for this
+example shows the window:
+
+.. image:: E1-L0-cw.png
+
+In this concurrency display, hovering the mouse over a task will
+bring up a "tooltip" that shows the clock time taken by that task,
+and if available, the source file name and line number of the code
+that started that task.
+
+In the main window, clicking on the red part of the line between
+locale 0 and locale 1 will produce a window that looks like:
+
+.. image:: E1-C1.0.png
+
+It is important to notice the direction of the *arrow* in the header
+for the windows.  This is for communication from locale 1 to locale 0.
+The total number of communication calls was 12.  It is further broken
+out into three components:
+
+  - *Gets:* This is a communication call initiated by locale 0 to
+    get a data located on locale 1.
+
+  - *Puts:* This is a communication call initiated by *locale 1*
+    to put data from locale 1 onto locale 0.
+
+  - *Forks:* This where *locale 1* starts a task running
+    on locale 0.  As part of the task start, a block of data is sent to
+    locale 0 as an argument to the task.  This data is considered a
+    communication call  by *chplvis*.
+
+
+
+
+Example 2
+---------
+
+In many programs, one will want to look at a number of small parts of
+their program in addition to seeing the total statistics.  prog2.chpl
+gives an example of using the *VisualDebug* functions
+*tagVdebug(name)* and *pauseVdebug()*.
+
+::
```

Comment:
This doesn't work with the standard rst2html.py tool.   I'm not sure which tool does this.